### ISSUE TITLE
Crash: java.lang.RuntimeException: MulticastLock under-locked react-native-udp

### ISSUE THREAD

**Issue by hubciorz** [tradle/react-native-udp#144]
## Description
App sometimes crashes after going to background (Android)

## Error
```java
com.tradle.react.UdpSockets$6.doInBackgroundGuarded
java.lang.RuntimeException: MulticastLock under-locked react-native-udp
```

## Stack traces
```java
android.net.wifi.WifiManager$MulticastLock.release WifiManager.java:4268
com.tradle.react.UdpSockets$6.doInBackgroundGuarded UdpSockets.java:273
com.tradle.react.UdpSockets$6.doInBackgroundGuarded UdpSockets.java:263
com.facebook.react.bridge.GuardedAsyncTask.doInBackground GuardedAsyncTask.java:36
com.facebook.react.bridge.GuardedAsyncTask.doInBackground GuardedAsyncTask.java:20
android.os.AsyncTask$3.call AsyncTask.java:378
java.util.concurrent.FutureTask.run FutureTask.java:266
java.util.concurrent.ThreadPoolExecutor.runWorker ThreadPoolExecutor.java:1167
java.util.concurrent.ThreadPoolExecutor$Worker.run ThreadPoolExecutor.java:641
java.lang.Thread.run Thread.java:919
```

## Steps to reproduce
Hard to say. It happens very rarely and it seems that you need to go to background at the right moment.

## Expected behavior
No crash, perhaps.

## Relevant information
|             |     |
| ----------- | --- |
| react-native | 0.62.1   |
| react-native-udp  | 4.0.3 (but happens on earlier versions, too)   |

## Occurred on
|             |     |
| ----------- | --- |
| Device | OS   |
| Xperia XZ2  | Android 10.0  |
| LG K4  | Android 6.0.1   |

**Comment by Rapsssito** [comment]
@hubciorz, thanks for your feedback! I don't know what is causing the error, but I am working on switching all multi-thread code to the Java standard API. It might fix your issue. I will create a PR which will close this issue when merged so you are notified.

**Comment by hubciorz** [comment]
@Rapsssito Good to hear that! Good luck and thank you.

**Comment by hubciorz** [comment]
@Rapsssito So far so good. No crashes. Tested by about 1000 users. I will let you know if there is any. Many thanks!

**Comment by hubciorz** [comment]
@Rapsssito It seems that crashes happen less frequently, but if they do, stack trace is quite different:

```java
com.tradle.react.UdpSockets$6.run
java.lang.RuntimeException: MulticastLock under-locked react-native-udp
```

```java
android.net.wifi.WifiManager$MulticastLock.release WifiManager.java:4291
com.tradle.react.UdpSockets$6.run UdpSockets.java:257
java.lang.Thread.run Thread.java:919
java.util.concurrent.ThreadPoolExecutor.runWorker ThreadPoolExecutor.java:1167
java.util.concurrent.ThreadPoolExecutor$Worker.run ThreadPoolExecutor.java:641
java.lang.Thread.run Thread.java:919
```

**Comment by Rapsssito** [comment]
@hubciorz could you provide me with an example code so I can reproduce the issue?

**Comment by hubciorz** [comment]
@Rapsssito This is just a standard udp listening code, as I pasted in #143 (btw, this one has been fixed). It's hard to reproduce the bug. A hint - going to background/foreground may be the key.

**Comment by hubciorz** [comment]
@Rapsssito I think it is likely to happen when someone loses wifi connection for a while. Please take a look a these comments: 
https://github.com/andriydruk/RxDNSSD/issues/59#issuecomment-457435808
https://github.com/andriydruk/RxDNSSD/issues/59#issuecomment-458074961

**Comment by Rapsssito** [comment]
@hubciorz thanks! However, I will take care of #147 first since crashes happen more frequently. I will come back to this one once that issue is fixed.

**Comment by Rapsssito** [comment]
Fixed by #148
